src/transfers-history/client.ts

```diff
@@ -0,0 +1,121 @@
+import EventEmitter from "events";
+import { clientConfig } from "./config";
+import { SpokePoolEventsQueryService } from "./services/SpokePoolEventsQueryService";
+import { Logger, LogLevel } from "./adapters/logger";
+import { providers } from "ethers";
+import { SpokePool, SpokePool__factory } from "@across-protocol/contracts-v2";
+import { ChainId } from "./adapters/web3/model";
+import { SpokePoolEventsQuerier } from "./adapters/web3";
+import { TransfersRepository } from "./adapters/db/transfers-repository";
+
+export enum TransfersHistoryEvent {
+  TransfersUpdated = "TransfersUpdated",
+}
+
+export type TransfersUpdatedEventListenerParams = {
+  depositorAddr: string;
+  filledTransfersCount: number;
+  pendingTransfersCount: number;
+};
+
+export type TransfersUpdatedEventListener = (params: TransfersUpdatedEventListenerParams) => void;
+export type TransfersHistoryClientEventListener = TransfersUpdatedEventListener;
+export type TransfersHistoryClientParams = {
+  chains: {
+    chainId: ChainId;
+    providerUrl: string;
+  }[];
+  pollingIntervalSeconds?: number;
+};
+
+export class TransfersHistoryClient {
+  private eventEmitter = new EventEmitter();
```

Comment:
> if you want to use event emitter, i think you could also inherit from it so the class natively supports events.

I wanted to keep it separated in case the events logic becomes more complex and want to migrate to a more complicated event bus/event broker approach

> also you could be less opinionated and just use a callback in the constructor, then the caller can decide if they need an event emitter or not ( or what kind they want)

@daywiss hmm, what if we will have multiple event types in the future? Add all of them to the constructor?
I kind of implemented this way on purpose. Events are a separated concern and I didn't way future events changes to impact the current function/constructor calls. One improvement I can do is instantiate the event emitter only if the user subscribes to events using `.on()` call

Reply:
callbacks would work teh same as emitting an event, theres an event type which can be an enum or a string, then theres some common data interface.  the user then can subscribe to particular event types using an event emitter.

```
const events = new Events()
// this is now a callback which the user can decide if they need an event emitter
const transferHistory = new TransferHistory(config,(type,data)=>events.emit(type,data))

// somewhere we want to subscribe
events.on('pendingTransfers',(transfer)=>// etc)